modules/apps/27-interchain-accounts/controller/migrations/v5/migrations.go

```diff
@@ -0,0 +1,58 @@
+package v5
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	capabilitykeeper "github.com/cosmos/cosmos-sdk/x/capability/keeper"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+
+	"github.com/cosmos/ibc-go/v5/modules/apps/27-interchain-accounts/controller/types"
+)
+
+// MigrateICS27ChannelCapability performs a search on a prefix store using the provided store key and module name.
+// It retrieves the associated channel capability index and reassigns ownership to the ICS27 controller submodule.
+func MigrateICS27ChannelCapability(
+	ctx sdk.Context,
+	cdc codec.BinaryCodec,
+	storeKey storetypes.StoreKey,
+	capabilityKeeper *capabilitykeeper.Keeper,
+	module string,
+) error {
+	// construct a prefix store using the x/capability index prefix: index->capability owners
+	prefixStore := prefix.NewStore(ctx.KVStore(storeKey), capabilitytypes.KeyPrefixIndexCapability)
+	iterator := sdk.KVStorePrefixIterator(prefixStore, nil)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		// unmarshal the capability index value and set of owners
+		index := capabilitytypes.IndexFromKey(iterator.Key())
+
+		var owners capabilitytypes.CapabilityOwners
+		cdc.MustUnmarshal(iterator.Value(), &owners)
+
+		for _, owner := range owners.GetOwners() {
+			if owner.Module == module {
+				// remove the owner from the set
+				owners.Remove(owner)
+
+				// reassign the owner module to icacontroller
+				owner.Module = types.SubModuleName
```

Comment:
do we need to remove, update, and re-add? Can we just do

```golang
owners[i].Module = types.SubModuleName
capabilityKeeper.SetOwners(ctx, index, owners)
```

Or will this not behave how I think it will?

Reply:
Seems like the APIs don't play nicely with this approach. I've tried rejigging the code a bit but I think its easier to leave as is 😅 

tldr; is the `CapabilityOwners` type encapsulates the slice and it makes it awkward to use index referencing